pkg/link/link_api.go

```diff
@@ -80,20 +96,12 @@ func getConfigureBaseURL() (string, error) {
 	return baseURL, nil
 }
 
-func parseUserBaseURLSelection(option string) (string, error) {
-  option = strings.TrimSpace(option)
-  switch option {
-  case "1":
-    return PRODUCTION_BASE_URL, nil
-  case "2":
-    return STAGING_BASE_URL, nil
-  case "3":
-    return TEST_BASE_URL, nil
-  case "4":
-    return getCustomBaseURL()
-  default:
-    return getCustomBaseURL()
+func parseUserBaseURLSelection(option int) (string, error) {
+  environment := ENVIRONMENTS[option]
```

Comment:
yeah I thought of that, and it does panic with a clear message. I was airing on the side of it not being that useful and that it panics so there's no silent unintended action. Given the intended users (mostly developers) If you're presented with something that says:
```
Choose the environment (enter a number between 0 and 3):
0 production
1 staging
2 test
3 custom
```
and you enter 4 then its kind of on you, but perhaps that's the wrong mindset :)

Having said that, I did just try adding a len() check and returning an error but I think the error handling isn't quite right because returning and error out of that function just fails silently and nothing is presented back to the user. Might just leave it to panic until that's fixed which I can look at on Tuesday?

Reply:
now the error handling is fixed, I've added an out of bounds check 👍